astropy/units/function/logarithmic.py

```diff
@@ -277,6 +277,62 @@ def __isub__(self, other):
         self._set_unit(new_unit)
         return self
 
+    def __mul__(self, other):
+        # Multiply by a float or a dimensionless quantity
+        try:
+            # Multiplying a log means putting the factor into the exponent
+            # of the unit
+            new_unit = self.unit._copy(self.unit.physical_unit**other)
```

Comment:
True. This probably happend by copy&pasting a template from somewhere else.

Reply:
My last comment was wrong, sorry. The point here is that I want to keep the LogUnit here. If I do just a
```Python
new_unit = self.unit.physical_unit**other
```
then the "Magnitude" is los, and this is converted to a quantity:
```
>>> m = 1*u.mag()
>>> m * 2
<Quantity 2.>
```

Reply:
Ah, yes, of course! Silly to have missed that. To avoid confusing future selves, how about
```
new_physical_unit = self.unit.physical ** other
result = ...
return self._new_view(result, self.unit._copy(new_physical_unit))
```

Reply:
Sure. Just pushed.